be/src/exprs/aggregate_functions.cpp

```diff
@@ -2362,8 +2363,37 @@ void AggregateFunctions::offset_fn_init(FunctionContext* ctx, T* dst) {
     DCHECK(ctx->is_arg_constant(1));
     DCHECK(ctx->is_arg_constant(2));
     DCHECK_EQ(ctx->get_arg_type(0)->type, ctx->get_arg_type(2)->type);
-    *dst = *static_cast<T*>(ctx->get_constant_arg(2));
+    T src = *static_cast<T*>(ctx->get_constant_arg(2));
+    // The literal null is sometimes incorrectly converted to int, so *dst = src may cause SEGV
```

Comment:
I'm not sure in which case this assign will cause SEGV. All Val is built-in type except StringVal, it seems that it can not lead segment fault.